clippy_lints/src/utils/mod.rs

```diff
@@ -293,7 +293,9 @@ pub fn qpath_res(cx: &LateContext<'_, '_>, qpath: &hir::QPath<'_>, id: hir::HirI
         hir::QPath::Resolved(_, path) => path.res,
         hir::QPath::TypeRelative(..) => {
             if cx.tcx.has_typeck_tables(id.owner.to_def_id()) {
-                cx.tcx.typeck_tables_of(id.owner.to_def_id()).qpath_res(qpath, id)
+                cx.tcx
+                    .typeck_tables_of(id.owner.to_def_id().expect_local())
```

Comment:
I think this is the same:

```suggestion
                    .typeck_tables_of(id.owner)
```